src/llama_stack_api/connectors.py

```diff
@@ -0,0 +1,171 @@
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from datetime import datetime
+from enum import StrEnum
+from typing import Any, Literal, Protocol
+
+from pydantic import BaseModel, Field
+from typing_extensions import runtime_checkable
+
+from llama_stack_api.resource import Resource, ResourceType
+from llama_stack_api.schema_utils import json_schema_type, webmethod
+from llama_stack_api.tools import ToolDef
+from llama_stack_api.version import LLAMA_STACK_API_V1ALPHA
+
+
+@json_schema_type
+class ConnectorType(StrEnum):
+    """Type of connector."""
+
+    MCP = "mcp"
+
+
+@json_schema_type
+class Connector(Resource):
+    """A connector resource representing a connector registered in Llama Stack.
+
+    :param type: Type of resource, always 'connector' for connectors
+    :param connector_type: Type of connector (e.g., MCP)
+    :param connector_id: User-specified identifier for the connector
+    :param url: URL of the connector
+    :param created_at: Timestamp of creation
+    :param updated_at: Timestamp of last update
+    :param server_label: (Optional) Label of the server
+    :param server_description: (Optional) Description of the server
+    :param tools: (Optional) List of tools available from the connector
+    :param registry_id: (Optional) ID of the registry this connector belongs to
+    """
+
+    model_config = {"populate_by_name": True}
+
+    type: Literal[ResourceType.connector] = ResourceType.connector
+    connector_type: ConnectorType = Field(default=ConnectorType.MCP)
+    user_connector_id: str | None = Field(
+        default=None, alias="connector_id", description="User-specified identifier for the connector"
+    )
+    url: str = Field(..., description="URL of the connector")
+    created_at: datetime = Field(..., description="Timestamp of creation")
+    updated_at: datetime = Field(..., description="Timestamp of last update")
+    server_name: str | None = Field(default=None, description="Name of the server")
+    server_label: str | None = Field(default=None, description="Label of the server")
+    server_description: str | None = Field(default=None, description="Description of the server")
+    # TODO: using ToolDef for now, but MCPListToolsTool should probably be updated and used instead
+    # once toolgroups are removed completely
+    tools: list[ToolDef] | None = Field(default=None, description="List of tools available from the connector")
+    registry_id: str | None = Field(default=None, description="ID of the registry this connector belongs to")
+
+    def _generate_connector_id(self) -> str:
+        name = self.server_name if self.server_name is not None else self.identifier
+        if self.registry_id is not None:
+            return f"{self.connector_type.value}::{self.registry_id}::{name}"
+        return f"{self.connector_type.value}::{name}"
+
+    @property
+    def connector_id(self) -> str:
```

Comment:
I don't think this should be a property -- the write path should handle setting an immutable `connector_id` during a _create_ operation only. You should not be able to change the identifier later either.